internal/payloads/error_payload.go

```diff
@@ -66,40 +69,89 @@ func PubkeyDuplicateError(ctx context.Context, message string, err error) *Respo
 	}
 }
 
-func ClientError(ctx context.Context, client string, message string, err error, status int) *ResponseError {
-	msg := fmt.Sprintf("%v error: %s: %v", client, message, err)
+func ClientErrorHelper(err error) (int, string) {
+	if errors.Is(err, clients.NotFoundErr) {
+		return 404, "service returned not found or no data"
+	} else if errors.Is(err, clients.UnauthorizedErr) {
+		return 401, "service returned unauthorized"
+	} else if errors.Is(err, clients.ForbiddenErr) {
+		return 403, "service returned forbidden"
+	} else if errors.Is(err, clients.Non2xxResponseErr) {
+		return 500, "service did not return 2xx"
+	}
+	return 0, ""
+}
+
+func SourcesErrorHelper(err error) (int, string) {
+	if errors.Is(err, httpClients.MoreThanOneAuthenticationForSourceErr) {
+		return 500, "sources returned more than one authentication"
+	} else if errors.Is(err, httpClients.ApplicationNotFoundErr) {
+		return 404, "sources application not found"
+	} else if errors.Is(err, httpClients.ApplicationTypeNotFoundErr) {
+		return 404, "unexpected source type"
+	} else if errors.Is(err, httpClients.SourceNotFoundErr) {
+		return 404, "source not found"
+	} else if errors.Is(err, httpClients.AuthenticationSourceAssociationErr) {
+		return 500, "authentication associated to source id not found"
+	} else if errors.Is(err, httpClients.AuthenticationForSourcesNotFoundErr) {
+		return 404, "authentication for source not found"
+	}
+	return 0, ""
+}
+
+func ImageBuilderHelper(err error) (int, string) {
+	if errors.Is(err, httpClients.ComposeNotFoundErr) {
+		return 404, "image builder did not find image compose"
+	} else if errors.Is(err, httpClients.ImageStatusErr) {
+		return 500, "image builder has not finished the build of requested image"
+	} else if errors.Is(err, httpClients.UnknownImageTypeErr) {
+		return 500, "unknown image type"
+	} else if errors.Is(err, httpClients.AmiNotFoundInStatusErr) {
+		return 404, "image builder did not find AMI in status"
+	} else if errors.Is(err, httpClients.NameNotFoundInStatusErr) {
+		return 404, "image builder did not find image name in image status"
+	} else if errors.Is(err, httpClients.IDNotFoundInStatusErr) {
+		return 404, "image builder did not find project id in image status"
+	}
+	return 0, ""
+}
```

Comment:
Do not export these functions outside of the package, unless they are used elsewhere, which I think they should not be.

Reviewers: We sort of landed on this solution we would like to test if it successful. The idea is to have a single place where we set all HTTP result codes and user-facing messages. If this turns out to be a success, we can refactor this into some types (maps perhaps) and make it more generic. For now this is only for HTTP clients.

Reply:
I like the approach :+1: